framework/components/useSearchParams.tsx

```diff
@@ -4,10 +4,15 @@ import { useWindowLocation } from './useWindowLocation';
 export function useSearchParams(): [URLSearchParams, (setSearchParams: URLSearchParams) => void] {
   const location = useWindowLocation();
   const pathname = location.location?.pathname || '/';
-  const searchParams = useMemo<URLSearchParams>(
-    () => new URLSearchParams(location.location?.search ?? '/'),
-    [location.location?.search]
-  );
+  const searchParams = useMemo<URLSearchParams>(() => {
+    /** Cypress component tests add a specPath param that must be ignored */
+    let search = location.location?.search;
+    if (search && search.includes('?specPath')) {
+      search = search.substring(0, search.indexOf('&specPath='));
```

Comment:
You search with a `?` in the if statement, but find indexOf with a `&`. Does Cypress always add this at the beginning of the query string, or do we need to check for it anywhere in the string?

Reply:
Thanks for catching that! Updated 👍